bucket.go

```diff
@@ -362,14 +367,18 @@ func (b *Bucket) MoveBucket(key []byte, dstBucket *Bucket) (err error) {
 	if !bytes.Equal(newKey, k) {
 		return errors.ErrBucketNotFound
 	} else if (flags & common.BucketLeafFlag) == 0 {
-		lg.Errorf("An incompatible key %s exists in the source bucket", string(newKey))
+		if lg != discardLogger {
+			lg.Errorf("An incompatible key %s exists in the source bucket", newKey)
+		}
```

Comment:
I suggest not to change this for better readability.  Usually it won't run into such case/branch, so there is no impact on the performance.